### PR TITLE
Pull in `libsodium` as a submodule and build from source

### PR DESCRIPTION
**Pull request by kasperisager**
This allows the module to participate in cross-compilation as it no longer relies on Autotools for building `libsodium`, but instead builds it directly using either GYP or CMake.

### CODE REVIEW

**Comment by kasperisager** [comment]
@mafintosh This is ready for review.

File: .gitignore [hunk 1]
```diff
@@ -1,9 +1,4 @@
+build
+prebuilds
```
Comment (mafintosh) [inline .gitignore:2]: you can't gitignore prebuilds, then npm wont publish them

Comment (kasperisager) [reply .gitignore:2]: npm publishes everything in `files` in `package.json` even if `.gitignored`'ed.

Comment (mafintosh) [reply .gitignore:2]: oh really, then ok

**Pull request merged**